File: app/Models/Producto.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Producto extends Model
{
    use HasFactory;

    protected $table = 'producto';
    protected $primaryKey = 'id_producto';

    protected $fillable = [
        'nombre_producto',
        'stock',
        'stock_min',
        'stock_max',
        'precio_compra',
        'precio_venta',
        'tipo_producto',
        'estado',
        'fecha_ingreso',
        'id_categoria'
    ];

    protected $casts = [
        'fecha_ingreso' => 'date',
        'precio_compra' => 'decimal:2',
        'precio_venta' => 'decimal:2',
    ];

    public function categoria()
    {
        return $this->belongsTo(CategoriaProducto::class, 'id_categoria', 'id_categoria');
    }

    public function movimientos()
    {
        return $this->hasMany(Movimiento::class, 'id_producto', 'id_producto');
    }

    public function unidadesMedida()
    {
        return $this->hasMany(UnidadMedida::class, 'id_producto', 'id_producto');
    }

    public function productosTerminados()
    {
        return $this->hasMany(ProductoTerminado::class, 'id_producto', 'id_producto');
    }

    public function promociones()
    {
        return $this->hasMany(Promocion::class, 'id_producto', 'id_producto');
    }

    public function pedidoProductos()
    {
        return $this->hasMany(PedidoProducto::class, 'id_producto', 'id_producto');
    }

    public function pedidos()
    {
        return $this->belongsToMany(Pedido::class, 'pedido_producto', 'id_producto', 'id_pedido')
                    ->withPivot('cantidad', 'precio_unitario', 'subtotal')
                    ->withTimestamps();
    }
}

File: src/main/java/com/Wurger/model/Producto.java
package com.Wurger.model;

import com.fasterxml.jackson.annotation.JsonIgnore; // Importante para evitar errores
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "producto")
@Getter
@Setter
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_producto")
    private Integer id;

    @Column(name = "nombre_producto", nullable = false, length = 100)
    private String nombreProducto;

    private Integer stock;

    @Column(name = "stock_min")
    private Integer stockMin;

    @Column(name = "stock_max")
    private Integer stockMax;

    @Column(name = "precio_compra")
    private BigDecimal precioCompra;

    @Column(name = "precio_venta")
    private BigDecimal precioVenta;

    @Column(name = "tipo_producto", length = 50)
    private String tipoProducto;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Estado estado;

    @Column(name = "fecha_ingreso")
    private LocalDate fechaIngreso;

    @Column(name = "imagen", length = 500)
    private String imagen;

    // --- RELACIONES ---
    @ManyToOne
    @JoinColumn(name = "id_categoria")
    private CategoriaProducto categoria;

    // Aquí SÍ ponemos JsonIgnore para que no traiga listas infinitas
    @OneToMany(mappedBy = "producto", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<UnidadMedida> unidades;

    @OneToMany(mappedBy = "producto", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Movimiento> movimientos;

    @OneToOne(mappedBy = "producto", cascade = CascadeType.ALL)
    @JsonIgnore
    private ProductoTerminado productoTerminado;

    @OneToMany(mappedBy = "producto", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Promocion> promociones;

    // --- ENUMS ---
    public enum Estado {
        Activo,
        Inactivo
    }
}